magpie/services.py

```diff
@@ -844,57 +913,119 @@ class ServiceGeoserverBase(ServiceOWS):
     Provides basic configuration parameters and functionalities shared by `Geoserver` implementations.
     """
 
-    @property
+    @classmethod
+    @classproperty
     @abc.abstractmethod
-    def resource_scoped(self):
+    def resource_scoped(cls):
         # type: () -> bool
         """
-        Indicates if the resource is allowed to employ scoped workspace naming.
+        Indicates if the :term:`Service` is allowed to employ scoped :class:`models.Workspace` naming.
 
         When allowed, the :class:`models.Workspace` can be inferred from the request parameter
         defined by :attr:`resource_param` to retrieve scoped name as ``<WORKSPACE>:<RESOURCE>``.
-        When not allowed, the resource name is left untouched can will not attempt to infer
-        any :class:`models.Workspace` from it.
+        When not allowed, the resource name is left untouched and `Magpie` will not attempt to infer
+        any :class:`models.Workspace` from it. In that case, :class:`models.Workspace` can only be specified
+        in the request path for isolated :term:`Resource` references.
+
+        .. note::
+            When this parameter is ``False`` for a given :term:`Service` implementation, children
+            :term:`Resources <Resource>` can still be named in a similar ``<namespace>:<element>`` fashion.
+            The only distinction is that the **full** :term:`Resource` should include this complete definition
+            instead of nesting ``<namespace>`` and ``<element>`` into two distinct :term:`Resources <Resource>`.
         """
         raise NotImplementedError
 
-    @property
+    @classproperty
     @abc.abstractmethod
-    def resource_param(self):
-        # type: () -> Str
+    def resource_multi(cls):  # noqa  # pylint: disable=E0213,no-self-argument
```

Comment:
Maybe my guess about wfs/wms was bad. Are we rather checking a request that involve multiple resources at the same time and thus all of them must be allowed or the request is deny. In which case, and it makes sense, the permission UI is not really affected.

Reply:
Yes, that's exactly it.
Technically the code allows many resource combinations across different services, but each service is usually selected and returns its own children resources.